src/Foundation/NSOperatingSystemVersion.cs

```diff
@@ -36,20 +36,102 @@ namespace Foundation  {
 	[SupportedOSPlatform ("macos")]
 	[SupportedOSPlatform ("tvos")]
 #endif
-	// NSProcessInfo.h
-	public struct NSOperatingSystemVersion {
+
+	public struct NSOperatingSystemVersion: IEquatable<NSOperatingSystemVersion>, IComparable{
 		public nint Major, Minor, PatchVersion;
 		
-		public NSOperatingSystemVersion (nint major, nint minor, nint patchVersion)
+        public NSOperatingSystemVersion (nint majorVersion, nint minorVersion, nint patchVersionArg)
+		{
+			Major = majorVersion;
+			Minor = minorVersion;
+			PatchVersion = patchVersionArg;
+		}
+
+        public NSOperatingSystemVersion (nint majorVersion, nint minorVersion)
 		{
-			Major = major;
-			Minor = minor;
-			PatchVersion = patchVersion;
+			Major = majorVersion;
+			Minor = minorVersion;
+			PatchVersion = 0;
 		}
 
+        public NSOperatingSystemVersion (nint majorVersion)
+		{
+			Major = majorVersion;
+			Minor = 0;
+			PatchVersion = 0;
+		}
+
+        public int CompareTo (NSOperatingSystemVersion otherVersion) {
+            int majorValue = Major.CompareTo(otherVersion.Major);
```

Comment:
```suggestion
            var majorValue = Major.CompareTo(otherVersion.Major);
```
we also tend to use var instead of the strong types when possible